src/redux/reducers/weatherReducer.js

```diff
@@ -20,6 +23,25 @@ const INITIAL_STATE = {
 
 const weatherReducer = (state = INITIAL_STATE, action) => {
   switch (action.type) {
+    case FETCH_WEATHER_START:
+      return {
+        ...state,
+        isFetching: true
+      };
+
+    case FETCH_WEATHER_SUCCESS:
+      return {
+        ...state,
+        isFetching: false
+      };
+
+    case FETCH_WEATHER_FAILURE:
+      return {
+        ...state,
+        isFetching: false,
+        isFetchingFailure: true
```

Comment:
Is there anywhere an action to reset isFetchingFailure to false back again...after an error occurred? Plus, I suggest to work with actual error object, because then you can manipulate with meaningful data, codes, messages inside. Not only boolean fact that error has just occurred